api/utils/aws/endpoint.go

```diff
@@ -130,6 +139,210 @@ func parseRedshiftCNEndpoint(endpoint string) (clusterID, region string, err err
 	return parts[0], parts[3], nil
 }
 
+// RedisEndpointInfo describes details extracted from a ElastiCache or MemoryDB
+// Redis endpoint.
+type RedisEndpointInfo struct {
+	// ID is the identifier of the endpoint.
+	ID string
+	// Region is the AWS region for the endpoint.
+	Region string
+	// TransitEncryptionEnabled specifies if in-transit encryption (TLS) is
+	// enabled.
+	TransitEncryptionEnabled bool
+	// EndpointType specifies the type of the endpoint.
+	EndpointType string
+}
+
+const (
+	// ElastiCacheConfigurationEndpoint is the configuration endpoint that used
+	// for cluster mode connection.
+	ElastiCacheConfigurationEndpoint = "configuration"
+	// ElastiCachePrimaryEndpoint is the endpoint of the primary node in the
+	// node group.
+	ElastiCachePrimaryEndpoint = "primary"
+	// ElastiCachePrimaryEndpoint is the endpoint of the replica nodes in the
+	// node group.
+	ElastiCacheReaderEndpoint = "reader"
+	// ElastiCacheNodeEndpoint is the endpoint that used to connect to an
+	// individual node.
+	ElastiCacheNodeEndpoint = "node"
+)
+
+// ParseElastiCacheRedisEndpoint extracts the details from the provided
+// ElastiCache Redis endpoint.
+//
+// https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/GettingStarted.ConnectToCacheNode.html
+func ParseElastiCacheRedisEndpoint(endpoint string) (*RedisEndpointInfo, error) {
```

Comment:
maybe just  `ParseElastiCacheEndpoint` to be consistent with the above convention where the `ElastiCache...` prefix is used.